tests/integration/components/component-dot-dollar-test.js

```diff
@@ -0,0 +1,35 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import Component from '@ember/component';
+import jQuery from 'jquery';
+
+const component = Component.extend({
+  didInsertElement() {
+    this.setJQueryElement(this.$(this.get('selector')));
+  }
+});
+
+module('Integration | Component | component-dot-dollar', function(hooks) {
+  setupRenderingTest(hooks);
+  hooks.beforeEach(function() {
+    this.owner.register('component:jquery-component', component);
+    this.$element = null;
+    this.setJQueryElement = ($) => this.$element = $;
+  });
+
+  test('it implements Component.$()', async function(assert) {
```

Comment:
The test does not check (yet) that no deprecation is raised. AFAICT we cannot use the test helpers in https://github.com/emberjs/ember.js/blob/master/packages/internal-test-helpers/lib/ember-dev/deprecation.ts, any other idea how to easily add test support here?

Reply:
Added deprecation checks!